v0.26 (#1437)
meilisearch/documentation: learn/configuration/instance_options.md

---
sidebarDepth: 2
---

# Configure Meilisearch at launch

You can configure Meilisearch at launch with **environment variables** and **command-line options**.

These startup options affect your entire Meilisearch instance, not just a single index. For settings that affect search within a single index, see [index settings](/learn/configuration/settings.md).

## Command-line options and flags

Pass **command-line options** and their respective values when launching a Meilisearch instance.

```bash
./meilisearch --db-path ./meilifiles --http-addr '127.0.0.1:7700'
```

In the previous example, `./meilisearch` is the command that launches a Meilisearch instance, while `--db-path` and `--http-addr` are options that modify this instance's behavior.

> Meilisearch also has a number of **command-line flags.** Unlike command-line options, **command-line flags don't take values**. If a flag is given, it is activated and changes Meilisearch's default behavior.

Reviewer: ```suggestion
Meilisearch also has a number of **command-line flags.** Unlike command-line options, **flags don't take values**. If a flag is given, it is activated and changes Meilisearch's default behavior.
```

```bash
./meilisearch --no-analytics
```

> The above command-line flag disables analytics for the Meilisearch instance without accepting any value.

Reviewer: ```suggestion
The above flag disables analytics for the Meilisearch instance and does not accept a value.
```
